### PR TITLE
mgr/cephadm: iscsi username and password defaults to admin

### PR DESCRIPTION
**Pull request by nizamial09**
A while ago, the iscsi username and password were defaulted to admin the config_dashboard part. But it was not saved to the `iscsi-gateway.cfg.j2` which results in a misconfiguration.

For eg, if you deploy an iscsi service with service_spec which doesn't have the api_user and api_password field it results the dashboard to show the gateways as always down because dashboard was unable to make request to the iscsi server and every api calls result in a `[dashboard ERROR rest_client] iscsi REST API failed GET req status: 401`.

Also fixed an issue with dashboard: 
```
                                              Traceback (most recent call last):
                                                File "/usr/share/ceph/mgr/dashboard/services/exception.py", line 47, in dashboard_exception_handler
                                                  return handler(*args, **kwargs)
                                                File "/lib/python3.6/site-packages/cherrypy/_cpdispatch.py", line 54, in __call__
                                                  return self.callable(*self.args, **self.kwargs)
                                                File "/usr/share/ceph/mgr/dashboard/controllers/_base_controller.py", line 263, in inner
                                                  ret = func(*args, **kwargs)
                                                File "/usr/share/ceph/mgr/dashboard/controllers/iscsi.py", line 140, in overview
                                                  result_images = self._get_images_info(config)
                                                File "/usr/share/ceph/mgr/dashboard/controllers/iscsi.py", line 151, in _get_images_info
                                                  tcmu_info = TcmuService.get_iscsi_info()
                                                File "/usr/share/ceph/mgr/dashboard/services/tcmu_service.py", line 86, in get_iscsi_info
                                                  TcmuService.remove_undetected_clients(images, daemons, daemon)
                                              UnboundLocalError: local variable 'daemon' referenced before assignment

```

Fixes: https://tracker.ceph.com/issues/57730
Signed-off-by: Nizamudeen A <nia@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by pereman2 — APPROVED** [review]

File: src/pybind/mgr/cephadm/services/iscsi.py [hunk 1]
```diff
@@ -67,6 +67,9 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
             })
 
         trusted_ip_list = self.get_trusted_ips(spec)
+        spec.api_user = spec.api_user or 'admin'
+        spec.api_password = spec.api_password or 'admin'
+        spec.api_port = spec.api_port or 5000
```
[the remark this answers is not in the document]
Comment (nizamial09) [reply src/pybind/mgr/cephadm/services/iscsi.py:72]: Yup, that makes sense to me. I'll move these changes to the initialization part. thanks for pointing those lines. (I honestly didn't knew which is the best place to make them default).

[the remark this answers is not in the document]
Comment (nizamial09) [reply src/pybind/mgr/cephadm/services/iscsi.py]: done!

**Review by adk3798 — APPROVED** [review]

File: src/python-common/ceph/deployment/service_spec.py [hunk 2]
```diff
@@ -928,9 +928,9 @@ def __init__(self,
                  service_id: Optional[str] = None,
                  pool: Optional[str] = None,
                  trusted_ip_list: Optional[str] = None,
-                 api_port: Optional[int] = None,
-                 api_user: Optional[str] = None,
-                 api_password: Optional[str] = None,
+                 api_port: Optional[int] = 5000,
+                 api_user: Optional[str] = 'admin',
+                 api_password: Optional[str] = 'admin',
```
Comment (adk3798) [inline src/python-common/ceph/deployment/service_spec.py:933]: there's a test `test_yaml` that seems to need a small adjustment after this
```
E       AssertionError: assert 'service_type...:1\n  - ::2\n' == 'service_type...:1\n  - ::2\n'
E         Skipping 84 identical leading characters in diff, use -v to show
E           ec:
E         +   api_password: admin
E         +   api_port: 5000
E             api_user: api_user
E             pool: pool
E             trusted_ip_list:...
E         
E         ...Full output truncated (3 lines hidden), use '-vv' to show

ceph/tests/test_service_spec.py:367: AssertionError
```

Comment (nizamial09) [reply src/python-common/ceph/deployment/service_spec.py:933]: thanks. updated!

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2022-12-02_21:56:44-orch:cephadm-wip-adk-testing-2022-12-02-0932-distro-default-smithi/

Failures tracked by:

- https://tracker.ceph.com/issues/49287
- https://tracker.ceph.com/issues/57755
- https://tracker.ceph.com/issues/57771
- https://tracker.ceph.com/issues/58145
- https://tracker.ceph.com/issues/58173

It's a lot of failures, but other than the the last one, they are all things we were aware of before the run (even the dead mgr-nfs-upgrade job had a `mount.nfs: mounting smithi174:/fake failed, reason given by server: No such file or directory` issue). The last one (https://tracker.ceph.com/issues/58173) as well seems to be a failure on a test setting flags on a pool, and given all the PRs in this run strictly touched mgr modules, I think it's unlikely any of the attached PRs caused the failures. Overall, I think the failures should only block merging on PRs related to nfs due to https://tracker.ceph.com/issues/58145 making it so we can't properly run our nfs related tests currently.

**Comment by adk3798** [comment]
jenkins retest this please

**Pull request merged**